worlds/ladx/LADXR/logic/overworld.py

```diff
@@ -61,9 +61,9 @@ def __init__(self, options, world_setup, r):
         self._addEntrance("banana_seller", sword_beach, banana_seller, r.bush)
         boomerang_cave = Location("Boomerang Cave")
         if options.boomerang == 'trade':
-            Location().add(BoomerangGuy()).connect(boomerang_cave, OR(BOOMERANG, HOOKSHOT, MAGIC_ROD, PEGASUS_BOOTS, FEATHER, SHOVEL))
+            Location().add(BoomerangGuy()).connect(boomerang_cave, AND(r.shuffled_magnifier), OR(BOOMERANG, HOOKSHOT, MAGIC_ROD, PEGASUS_BOOTS, FEATHER, SHOVEL))
```

Comment:
Yeah looking at the function signature of .connect(), it doesn't look like it handles a third arg (although it does eat it with `*` for some reason), so it would just be lost. And looking at every other instance of `AND()`, it has two or more inputs

Reply:
Yep, you're right, pushed the fix